engine/src/test/java/com/redhat/cloud/notifications/events/EndpointProcessorTest.java

```diff
@@ -318,4 +282,104 @@ void testEndpointRestrictionAccordingEventType(final boolean isEventTypeRestrict
+        regularEndpoint.setType(EndpointType.WEBHOOK);
+
+        final Endpoint systemEndpoint = new Endpoint();
+        systemEndpoint.setId(endpointUuid2);
+        systemEndpoint.setType(EndpointType.EMAIL_SUBSCRIPTION);
+
+        Action action = buildAction(orgId);
+
+        final EventType eventType = new EventType();
+        eventType.setId(UUID.randomUUID());
+
+        Event event = new Event();
+        event.setId(UUID.randomUUID());
+        event.setEventWrapper(new EventWrapperAction(action));
+        event.setEventType(eventType);
+        event.setOrgId(orgId);
+
+        Mockito.when(this.endpointRepository.getTargetEndpoints(Mockito.anyString(), Mockito.any(EventType.class))).thenReturn(List.of(regularEndpoint, blacklistedEndpoint, systemEndpoint));
+        Mockito.when(this.endpointRepository.getTargetEndpointsWithoutUsingBgs(Mockito.anyString(), Mockito.any(EventType.class))).thenReturn(List.of(regularEndpoint, blacklistedEndpoint, systemEndpoint));
+
+        Mockito.doNothing().when(this.webhookProcessor).process(Mockito.any(Event.class), Mockito.anyList());
+
+        Mockito.when(engineConfig.isBlacklistedEndpoint(eq(endpointUuid))).thenReturn(true);
+
+        this.endpointProcessor.process(event);
+
+        Mockito.verify(this.endpointRepository, Mockito.times(0)).findByUuidAndOrgId(endpointUuid, orgId);
+        Mockito.verify(this.webhookProcessor, Mockito.times(1)).process(eq(event), Mockito.anyList());
+        Mockito.verify(this.emailConnectorProcessor, Mockito.times(1)).process(eq(event), Mockito.anyList(), anyBoolean());
+        Mockito.verify(this.engineConfig, Mockito.times(2)).isBlacklistedEndpoint(any(UUID.class));
+
+        if (useEndpointToEventTypeDirectLink) {
+            Mockito.verify(this.endpointRepository, Mockito.times(1)).getTargetEndpointsWithoutUsingBgs(Mockito.anyString(), Mockito.any(EventType.class));
+            Mockito.verify(this.endpointRepository, Mockito.times(0)).getTargetEndpoints(Mockito.anyString(), Mockito.any(EventType.class));
+        } else {
+            Mockito.verify(this.endpointRepository, Mockito.times(1)).getTargetEndpoints(Mockito.anyString(), Mockito.any(EventType.class));
+            Mockito.verify(this.endpointRepository, Mockito.times(0)).getTargetEndpointsWithoutUsingBgs(Mockito.anyString(), Mockito.any(EventType.class));
+        }
+
+        // all endpoints blacklisted
+        Mockito.reset(this.webhookProcessor);
+        Mockito.reset(this.emailConnectorProcessor);
+        Mockito.reset(this.engineConfig);
+        Mockito.reset(this.engineConfig);
+
+        Mockito.when(this.endpointRepository.getTargetEndpoints(Mockito.anyString(), Mockito.any(EventType.class))).thenReturn(List.of(regularEndpoint, blacklistedEndpoint));
+        Mockito.when(this.endpointRepository.getTargetEndpointsWithoutUsingBgs(Mockito.anyString(), Mockito.any(EventType.class))).thenReturn(List.of(regularEndpoint, blacklistedEndpoint));
```

Comment:
I suppose you didn't include the email endpoint here because it's already tested above. Is that correct?

Reply:
I didn't incuded email endpoint because it is configured as system endpoint (org_id is null).
Such system endpoints can't be blacklisted and I test the case where every endpoints are blacklisted.